src/terminal/parser/stateMachine.cpp

```diff
@@ -1230,7 +1232,13 @@ bool StateMachine::FlushToTerminal()
     //      that pwchCurr was processed.
     // However, if we're here, then the processing of pwchChar triggered the
     //      engine to request the entire sequence get passed through, including pwchCurr.
-    return _engine->ActionPassThroughString(_run);
+    bool succeeded = true;
+    if (succeeded && _cachedSequence.has_value())
+    {
+        succeeded = _engine->ActionPassThroughString(*_cachedSequence);
+        _cachedSequence.reset();
+    }
+    return succeeded && _engine->ActionPassThroughString(_run);

```

Comment:
You're being tricky here and it's taking more than instantaneous to figure out what you're doing. Can you please add more comments to explain your trickery?

(Also that first `succeeded` test is weird given it's always true...)

Reply:
the first `succeeded` test is in case anybody ever adds another thing that might fail in this flow ;P but yeah, fair. I'll comment it